### PR TITLE
Add docs comparing Airflow and dbt concepts

### PR DESCRIPTION
**Pull request by tatiana**
Recently @ReadytoRocc suggested we had a documentation comparing dbt and Airflow concepts. This PR aims to address this gap. Any feedback and reviews would be very welcome since I'm still learning about dbt myself :)

### CODE REVIEW

File: docs/getting_started/dbt-airflow-concepts.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+| Airflow naming | dbt naming   | Description                                                                     | Differences                                                                 | References                                                                           |
++================+==============+=================================================================================+=============================================================================+======================================================================================+
+| DAG            | Workflow     | Pipeline (Direct Acyclic Graph) that contains a group of steps                  | Airflow expects upstream tasks to have passed to run downstream tasks.      | https://airflow.apache.org/docs/apache-airflow/2.7.1/core-concepts/dags.html         |
+|                |              |                                                                                 | dbt can run a subset of tasks assuming upstream tasks were run.             | https://docs.getdbt.com/docs/introduction                                            |
```
Comment (harels) [inline docs/getting_started/dbt-airflow-concepts.rst:18]: I would maybe put this difference in the task section?

Comment (harels) [reply docs/getting_started/dbt-airflow-concepts.rst:18]: The difference talks about airflow task dependencies, and this is discussing DAGs

File: docs/getting_started/dbt-airflow-concepts.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
++----------------+--------------+---------------------------------------------------------------------------------+-----------------------------------------------------------------------------+--------------------------------------------------------------------------------------+
+| Task           | Node         | Step within a pipeline (DAG or workflow)                                        | In dbt, users write mostly SQL and YML to define the steps of a pipeline.   | https://docs.getdbt.com/reference/node-selection/syntax                              |
```
Comment (harels) [inline docs/getting_started/dbt-airflow-concepts.rst:20]: I would add a row about languages and add the difference there

File: docs/getting_started/dbt-airflow-concepts.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+|                |              |                                                                                 |                                                                             | https://airflow.apache.org/docs/apache-airflow/2.7.1/core-concepts/variables.html    |
++----------------+--------------+---------------------------------------------------------------------------------+-----------------------------------------------------------------------------+--------------------------------------------------------------------------------------+
+| Templating     | Macros       | Jinja templating used to access variables, configuration and reference steps    | dbt encourages using jinja templating for control structures (if and for).  | https://docs.getdbt.com/docs/build/jinja-macros                                      |
+|                |              |                                                                                 | Airflow usage is limited to variables, macros and filters                   | https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html             |
```
Comment (harels) [inline docs/getting_started/dbt-airflow-concepts.rst:27]: I would say here that because Airflow is Python based, templating is supported natively

**Review by harels — APPROVED** [review]
left some notes that I think would be helpful

File: docs/getting_started/dbt-airflow-concepts.rst (hunk 1 above)
Comment (tatiana) [reply docs/getting_started/dbt-airflow-concepts.rst:18]: @harels I see tasks as individual nodes (vertices), whereas DAGs represent a group of nodes and edges - and how the nodes relate to each other - so I believe this comment is more suitable in the DAGs/workflows. WDYT? If you feel strongly about it, we can still change before the release 1.1.1.

**Pull request merged**